packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -20,54 +20,58 @@ import '../runner/flutter_command.dart';
 /// Map from package name to package version, used to artificially pin a pub
 /// package version in cases when upgrading to the latest breaks Flutter.
 const Map<String, String> _kManuallyPinnedDependencies = <String, String>{
-  // Add pinned packages here.
+  // Add pinned packages here. Please leave a comment explaining why.
+  // PACKAGES WITH INCOMPATIBLE LATER VERSIONS
   // Dart analyzer does not catch renamed or deleted files.
   // Therefore, we control the version of flutter_gallery_assets so that
   // existing tests do not fail when the package has a new version.
   'flutter_gallery_assets': '^0.2.0',
-  'mockito': '4.1.1',  // Prevent mockito from upgrading to the source gen version.
-  'vm_service_client': '0.2.6+2', // Final version before being marked deprecated.
   'flutter_template_images': '1.0.1', // Must always exactly match flutter_tools template.
-  'shelf': '0.7.5',
-  // Dart team owned nnbd deps
-  'archive': '3.0.0-nullsafety.0',
-  'async': '2.5.0-nullsafety.3',
-  'boolean_selector': '2.1.0-nullsafety.3',
-  'characters': '1.1.0-nullsafety.5',
-  'charcode': '1.2.0-nullsafety.3',
-  'clock': '1.1.0-nullsafety.3',
-  'collection': '1.15.0-nullsafety.5',
-  'intl': '0.17.0-nullsafety.2',
-  'fake_async': '1.2.0-nullsafety.3',
-  'js': '0.6.3-nullsafety.3',
-  'matcher': '0.12.10-nullsafety.3',
-  'meta': '1.3.0-nullsafety.6',
-  'path': '1.8.0-nullsafety.3',
-  'pedantic': '1.10.0-nullsafety.3',
-  'pool': '1.5.0-nullsafety.3',
-  'source_maps': '0.10.10-nullsafety.3',
-  'source_map_stack_trace': '2.1.0-nullsafety.4',
-  'source_span': '1.8.0-nullsafety.4',
-  'stack_trace': '1.10.0-nullsafety.6',
-  'stream_channel': '2.1.0-nullsafety.3',
-  'string_scanner': '1.1.0-nullsafety.3',
-  'term_glyph': '1.2.0-nullsafety.3',
-  'test': '1.16.0-nullsafety.16',
-  'test_api': '0.2.19-nullsafety.6',
-  'test_core': '0.3.12-nullsafety.15',
-  'typed_data': '1.3.0-nullsafety.5',
-  'vector_math': '2.1.0-nullsafety.5',
-  // Flutter team owned nnbd deps
-  'platform': '3.0.0-nullsafety.4',
-  'file': '6.0.0-nullsafety.4',
-  'process': '4.0.0-nullsafety.4',
-  'process_runner': '4.0.0-nullsafety.5',
+  'mockito': '4.1.1', // Prevent mockito from upgrading to the source gen version.
```

Comment:
Doesn't have to be this PR, but this may need to get updated to 4.1.4 at some point before long, since this is going to run afoul of the analyzer dependency.

Reply:
definitely not gonna try to upgrade mockito mixed in with other changes, that should definitely be in its own PR. :-)